packages/lesswrong/server/resolvers/dialogueChecksResolvers.ts

```diff
@@ -0,0 +1,38 @@
+import DialogueChecks from "../../lib/collections/dialogueChecks/collection";
+import { randomId } from "../../lib/random";
+import { augmentFieldsDict } from "../../lib/utils/schemaUtils";
+import { defineMutation } from "../utils/serverGraphqlUtil";
+
+defineMutation({
+  name: "upsertUserDialogueCheck",
+  resultType: "DialogueCheck",
+  argTypes: "(targetUserId: String!, checked: Boolean!)",
+  fn: async (_, {targetUserId, checked}:{targetUserId:string, checked:boolean}, {currentUser, repos}) => {
+    if (!currentUser) throw new Error("No check user was provided")
+    if (!targetUserId) throw new Error("No target user was provided")
+    const existingCheck = await DialogueChecks.findOne({targetUserId, userId: currentUser._id})
+    const id = existingCheck ? existingCheck._id : randomId()
+    const resposne = await repos.dialogueChecks.upsertDialogueCheck(id, currentUser._id, targetUserId, checked)
+    return {
+      _id: id,
+      userId: currentUser._id,
+      targetUserId,
+      checked,
+    }
+  } 
+})
+
+augmentFieldsDict(DialogueChecks, {
+  match: {
+    resolveAs: {
+      fieldName: 'match',
+      type: 'Boolean',
+      resolver: async (check: DbDialogueCheck, args: void, context: ResolverContext): Promise<Boolean> => {
```

Comment:
should return `Promise<boolean>` rather than `Promise<Boolean>` (those are actually different things; in this case the difference is irrelevant but in general don't want to be using the other form unnecessarily)

Reply:
ack